src/preamble.js

```diff
@@ -933,9 +943,9 @@ function updateGlobalBufferViews() {
   Module['HEAPF64'] = HEAPF64 = new Float64Array(buffer);
 }
 
-var STATIC_BASE = 0, STATICTOP = 0, staticSealed = false; // static area
```

Comment:
Sorry, missed that one, a good question. The pattern here is that the initialization to ` = 0;` would kill the value that these are initialized to earlier, i.e. in `pthread-main.js` there's

    var STATIC_BASE = 0;
    ...
    STATIC_BASE = <value received from main thread>;
    importScripts('a.js');

so if the file `a.js` has

    var STATIC_BASE = 0;

it would unconditionally override the value that `pthread-main.js` has already intialized it to. `var STATIC_BASE;` doesn't override, but just re-declares and keeps the value if it exists, or if it didn't, it defines the variable and sets it to `undefined`. (at least that's my understanding of the semantics)

I thought of doing the initialization explicitly on the main thread like this:

    var STATIC_BASE, STATICTOP, staticSealed; // static area
    var STACK_BASE, STACKTOP, STACK_MAX; // stack area
    var DYNAMIC_BASE, DYNAMICTOP_PTR; // dynamic area handled by sbrk
    if (!ENVIRONMENT_IS_PTHREAD) {
        STATIC_BASE = STATICTOP = STACK_BASE = STACKTOP = STACK_MAX = DYNAMIC_BASE = DYNAMICTOP_PTR = 0;
        staticSealed = false;
    }

but when running the code without these initialization assignments, I realized everything passes even without these, so figured to just leave it in the kind of "why pay code lines for statements that aren't needed?" fashion. If you think the init to zero is worth having in, certainly can add those. What do you think?